nearcache_map.go

```diff
@@ -62,8 +84,52 @@ func newNearCacheMap(nc *inearcache.NearCache, ncc *nearcache.Config, ss *serial
 	return ncm, nil
 }
 
-func (ncm *nearCacheMap) registerInvalidationListener() {
-	fmt.Println("IMPLEMENT ME: registerInvalidationListener")
+func (ncm *nearCacheMap) Destroy(ctx context.Context, name string) error {
+	ncm.lg.Trace(func() string {
+		return fmt.Sprintf("hazelcast.nearCacheMap.Destroy: %s", name)
+	})
+	s := ncm.invalidationListenerID.Load()
+	if s == nil {
+		return nil
+	}
+	// removeNearCacheInvalidationListener
+	sid := s.(types.UUID)
+	ncm.rt.DeregisterHandler(name)
+	return ncm.lb.Remove(ctx, sid)
+}
+
+func (ncm *nearCacheMap) registerInvalidationListener(ctx context.Context, name string, local bool) error {
+	// port of: com.hazelcast.client.map.impl.nearcache.NearCachedClientMapProxy#registerInvalidationListener
+	sid := types.NewUUID()
```

Comment:
I believe it will be easier to read if you move sid variable to a place closer to where it is used. (above `removeMsg := codec.EncodeMapRemoveEntryListenerRequest(name, sid)`)

Reply:
Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/807/commits/0d49f98e13b8fc0d4843db1366d353fb6a96a174